backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/LlvmModuleSpecificationImpl.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.kotlin.backend.konan
+
+import org.jetbrains.kotlin.backend.konan.descriptors.konanLibrary
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
+import org.jetbrains.kotlin.ir.declarations.IrDeclaration
+import org.jetbrains.kotlin.ir.declarations.IrModuleFragment
+import org.jetbrains.kotlin.ir.util.module
+import org.jetbrains.kotlin.konan.library.KonanLibrary
+
+// TODO: consider making two implementations: for producing cache and anything else.
+internal class LlvmModuleSpecificationImpl(
+        private val cachedLibraries: CachedLibraries,
+        private val producingCache: Boolean,
+        private val librariesToCache: Set<KonanLibrary>
```

Comment:
AFAIU, It's always either `producingCache && librariesToCache.isNotEmpty()` or `!producingCache && librariesToCache.isEmpty()`. Maybe remove `producingCache` field?

Reply:
Correct, but 
1. I prefer this to be more explicit.
2. I consider making two implementation of `LlvmModuleSpecification`, and `producingCache` reflects this.